### DISCUSSION TITLE
Color scale bar in gene_score/expression tsne/umap plot, changing Min-Max to 0-Max

### DISCUSSION THREAD

**Discussion by willey2020** [GreenleafLab/ArchR#579] [Questions / Documentation]
Hello! Thanks so much for this great package!

**Describe the problem that your feature request would address.**

Currently, the gene score or gene integration matrix color showing on the tsne/umap plot has a color bar, using the scale of "Min/Max". I found that this is very useful to demonstrate the gene behavior in an "on-off" fashion, like 0.01 to 5 or 5 to 0.01. Min is close to 0. 

However, for genes that do not fully turn on or turn off, but just have upregulation or downregulation from like 3 to 6 or 5 to 2, the color bar using Min in very dark black color gives a kind of impression to the reader that this gene is still showing switching on-off fashion, not representing an upregulation from a certain amount of initial amount of activity/expression which is not closed to 0, or a downregulation to a certain amount which is not 0 too. 

**Describe the solution you'd like**
To clearly demonstrate these types of gene behaviors, in plotembeding function, is it possible to show the color bar with a scale from 0 to a certain maximum for each gene, but not MIN to maximum? For example,  a gene changes from 3 to 6 but not 0.01 to 5, whose color for the cells having 3 value will not be fully black in the Min fashion, but dim yellow which will represent its decent level.   
Thanks very much!

**Comment by rcorces** [comment]
Short answer - yes this is possible

My 2 cents on your rationale - I would strongly caution against the logic that a gene score value of 0.5 indicates low expression for Gene A and a gene score value of 2 indicates high expression for Gene B. Gene scores are inferred based on local accessibility but the magnitude of the gene score is dependent on a lot of things that may or may not be relevant when comparing across genes. My interpretation is that using Min and Max is the most fair way of displaying these gene scores.

The full answer to you question is that you can pass additional paramters to `plotEmbedding()` to control plot attributes used in the underlying `ggPoint()` and `ggHex()` functions. This is indicated by the `...` in the function description:
https://www.archrproject.com/reference/plotEmbedding.html

Closing but feel free to comment if you dont think that this answered your question and I will re-open.

**Comment by willey2020** [comment]
Thank you so much! I successfully change the color scale through the way as you suggested. Using colorLimit in ggpoint. I can change it the min to 0, using colorLimit =c(0,3), It works well to change the color scale, however, I have to give a fixed max number, such as 3, or 4... in it to fix, could I ask how I can change the 3 to just the Max for each gene and it will automatically sense the max for each gene during the plot, but not a fixed number like 3? Thank you again!

**Reply by rohitrrj** [reply]
@willey2020 Hi, Can you provide a brief example of how you were able to get the colorLimits working, for me it seems that whatever limits I provide they seem to be always ignored while plotting.

**Reply by willey2020** [reply]
@rohitrrj , sorry very much for late reply, I didn't check the github account for a long time. I hope you have already solved the issue, but if not, 
I think you need to add "plotAs = 'points'"(or whatever any kind of shape you like for the cell point)
without "plotAs = 'points'", plotEmbedding seems to be ignoring the colorLimit=...

Please let me know if it doesn't work.

**Comment by rcorces** [comment]
Hmmm. That is a good question and I'm not sure there is a straightforward answer. The way I see it, if you are using impute weights for plotting gene scores, it is not straightforward to determine what the maximum and minimum values will be after applying the impute weights before you call `plotEmbedding()`.

If you weren't using impute weights, then you could just manually determine the max value from the gene score matrix and pass that.

Maybe @jgranja24 has thoughts?
